### PR TITLE
Adds possibility to disable maximization button on a perspective

### PR DESCRIPTION
**Pull request by paulovmr**

### CODE REVIEW

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayout.java [hunk 1]
```diff
@@ -78,14 +78,16 @@ void resizeTo(int width,
...
-     * <p>
+     * <p/>
      * <b>Important:</b> this feature is used by panels to unmaximize themselves. You should not pass a WorkbenchPanelView
      * to this method yourself; instead, you should use the panel's own API to unmaximize it. You are free to use this method
      * to unmaximize your own widgets that have previously been passed to {@link #maximize(Widget)}.
      * @param w the Widget to restore to its original size and location.
      */
     void unmaximize(Widget w);
 
+    void disableMaximization();
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayout.java]: @javadoc

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/WorkbenchPanelPresenter.java [hunk 2]
```diff
@@ -140,6 +140,8 @@ public void changeTitle(final PartDefinition part,
      */
     public void unmaximize();
 
+    void disableMaximization();
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/WorkbenchPanelPresenter.java]: @javadoc

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/impl/AbstractWorkbenchPanelView.java [hunk 3]
```diff
@@ -121,12 +121,17 @@ public Widget getPartDropRegion() {
...
 
     @Override
+    public void disableMaximization() {
+        layoutSelection.get().disableMaximization();
+    }
+
+    @Override
     public void maximize() {
-        layoutSelection.get().maximize(this);
+        layoutSelection.get().maximize(asWidget());
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/impl/AbstractWorkbenchPanelView.java]: are u sure?

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/impl/AbstractWorkbenchPanelView.java [hunk 4]
```diff
@@ -121,12 +121,17 @@ public Widget getPartDropRegion() {
...
     }
 
     @Override
     public void unmaximize() {
-        layoutSelection.get().unmaximize(this);
+        layoutSelection.get().unmaximize(asWidget());
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/impl/AbstractWorkbenchPanelView.java]: are u sure?

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/impl/AbstractSimpleWorkbenchPanelView.java [hunk 5]
```diff
@@ -145,6 +145,11 @@ public void unmaximize() {
     }
 
     @Override
+    public void disableMaximization() {
+        listBar.getMaximizeButton().setVisible(false);
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/panels/impl/AbstractSimpleWorkbenchPanelView.java]: disableMaximization()?

**Comment by jhrcek** [comment]
Shouldn't we fix the bug that maximization button sometimes breaks layout (reported here https://issues.jboss.org/browse/RHBPMS-4894) than to implement this kind of workarounds?

**Comment by paulovmr** [comment]
Hi @jhrcek , you are right, I haven't seem this ticket before... I will work on a more final solution and open another PR. Closing this in the meanwhile.

**Pull request closed without merging**